store/modules/forecastViz.js

```diff
@@ -52,6 +66,9 @@ export default moduleOptions => ({
       }
       this.dispatch('forecastViz/fetch_as_of_truth');
       this.dispatch('forecastViz/fetch_forecasts');
```

Comment:
I think we need to add `await` on these action dispatches now that we are not directly updating the truth source. When I ran this locally, the data and forecasts did not reliably update when i navigated forward and backward across as of dates.  When making this change, we will also have to make the function `async` (i.e., declare it as `async increment_as_of(state) {` above)

I actually think that both here and elsewhere where we are issuing multiple data fetches maybe we should instead use `Promise.all()` as described in this blog post: https://www.seanmcp.com/articles/await-multiple-promises-in-javascript/

```suggestion
      await this.dispatch('forecastViz/fetch_as_of_truth');
      await this.dispatch('forecastViz/fetch_forecasts');
```